src/components/app/ListOfPublishedProfiles.js

```diff
@@ -283,19 +283,33 @@ type State = {|
 |};
 
 export class ListOfPublishedProfiles extends PureComponent<Props, State> {
+  _isMounted = false;
+
   state = {
     profileDataList: null,
   };
 
   async componentDidMount() {
-    const profileDataList = await listAllProfileData();
+    this._isMounted = true;
+    await this.loadAllProfileData();
+    window.addEventListener('focus', this.loadAllProfileData);
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+    window.removeEventListener('focus', this.loadAllProfileData);
+  }
 
-    // It isn't ideal to use a setState here, but this is the only way.
-    // eslint-disable-next-line react/no-did-mount-set-state
-    this.setState({
-      // We want to display the list with the most recent uploaded profile first.
-      profileDataList: profileDataList.reverse(),
-    });
+  loadAllProfileData = async () => {
+    const profileDataList = await listAllProfileData();
+    if (this._isMounted) {
+      // It isn't ideal to use a setState here, but this is the only way.
+      // eslint-disable-next-line react/no-did-mount-set-state
```

Comment:
Actually this eslint rule doesn't seem to find the issue now, because it was moved in a different method. That's one of the eslint problems reported: that this rule suppression doesn't do anything. Can you please remove this `eslint-disable-next-line`?